src/com/palmergames/bukkit/towny/db/ObjectSaveUtil.java

```diff
@@ -0,0 +1,352 @@
+package com.palmergames.bukkit.towny.db;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.bukkit.Location;
+
+import com.google.gson.Gson;
+import com.palmergames.bukkit.towny.TownyMessaging;
+import com.palmergames.bukkit.towny.object.Nation;
+import com.palmergames.bukkit.towny.object.PermissionData;
+import com.palmergames.bukkit.towny.object.PlotGroup;
+import com.palmergames.bukkit.towny.object.Resident;
+import com.palmergames.bukkit.towny.object.Town;
+import com.palmergames.bukkit.towny.object.TownBlock;
+import com.palmergames.bukkit.towny.object.TownyObject;
+import com.palmergames.bukkit.towny.object.TownyWorld;
+import com.palmergames.bukkit.towny.object.jail.Jail;
+import com.palmergames.bukkit.towny.object.metadata.DataFieldIO;
+import com.palmergames.util.StringMgmt;
+
+public class ObjectSaveUtil {
+
+	/*
+	 * HashMap methods used to save objects in the TownyFlatFileSource and TownySQLSource
+	 */
+
+	public static HashMap<String, Object> getJailMap(Jail jail) throws Exception {
+		try {
+			HashMap<String, Object> jail_hm = new HashMap<>();
+			jail_hm.put("uuid", jail.getUUID());
+			jail_hm.put("townBlock", getTownBlockForSaving(jail.getTownBlock()));
+			
+			StringBuilder jailCellArray = new StringBuilder();
+			if (jail.hasCells())
+				for (Location cell : new ArrayList<>(jail.getJailCellLocations()))
+					jailCellArray.append(parseLocationForSaving(cell)).append(";");
+
+			jail_hm.put("spawns", jailCellArray);
+			
+			return jail_hm;
+		} catch (Exception e) {
+			TownyMessaging.sendErrorMsg("Saving: Jail HashMap could not be made.");
+			e.printStackTrace();
+			throw new Exception(e);
```

Comment:
I think it'd be best to remove the sendErrorMessage and printStackTrace and let the catcher do that, instead of a plain exception we could also wrap this into a new ObjectSaveException